### PR TITLE
Added RetryContext

### LINKED ISSUE #594: Add support for RetryContext
Issue: Azure/azure-webjobs-sdk#2595 added support to access retry context within execution context.

This tracking item to expose the same property in python functions. Currently blocked on Azure/azure-functions-host#7048

### PR DESCRIPTION
**Pull request by michaelpeng36**
### Issue describing the changes in this PR

Resolves #594.

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **should not** be added to the release notes for the next release
  * [x] Otherwise: I've added my notes to `release_notes.md`
* [x] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

**Comment by michaelpeng36** [comment]
I am planning on backporting these changes to the `v4.x/ps7.0`, `v3.x/ps7`, and `v3.x/ps6`. @Francisco-Gamino and @AnatoliB , let me know if backporting to these branches makes sense to you.

**Comment by Francisco-Gamino** [comment]
> I am planning on backporting these changes to the `v4.x/ps7.0`, `v3.x/ps7`, and `v3.x/ps6`. @Francisco-Gamino and @AnatoliB , let me know if backporting to these branches makes sense to you.

Thank you @michaelpeng36 for sending this out!

We should only port this change to `v4.x/ps7.0`, `v3.x/ps7`. Do you agree @AnatoliB?

File: release_notes.md [hunk 1]
```diff
@@ -1 +1,2 @@
-* Bug fix: Activity Functions can now use output bindings (https://github.com/Azure/azure-functions-powershell-worker/issues/646)
```
Comment (Francisco-Gamino) [inline release_notes.md]: We do not need to update this file. Please remove it from the PR.

**Comment by AnatoliB** [comment]
> We should only port this change to `v4.x/ps7.0`, `v3.x/ps7`. Do you agree @AnatoliB Anatoli Beliaev FTE?

I agree, no need to port to `/ps6`.

File: src/Public/RetryContext.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using Microsoft.Azure.WebJobs.Script.Grpc.Messages;
+
```
Comment (Francisco-Gamino) [inline src/Public/RetryContext.cs:7]: nit: Please remove the extra space.

File: src/Utility/TypeExtensions.cs [hunk 3]
```diff
@@ -153,9 +153,9 @@ internal static RpcException ToRpcException(this Exception exception)
         {
```
Comment (Francisco-Gamino) [inline src/Utility/TypeExtensions.cs:153]: Did you change anything on this file beside the order or parameters?

**Review by Francisco-Gamino — CHANGES_REQUESTED** [review]

**Comment by Francisco-Gamino** [comment]
@michaelpeng36 -- I left a couple of minor comments. 

Adding the `$RetryContext` variable will required the templates to be updated. Could you please open an issue at https://github.com/Azure/azure-functions-templates/issues for this? Thanks! 

/cc @stefanushinardi

File: src/Utility/TypeExtensions.cs (hunk 3 above)
Comment (michaelpeng36) [reply src/Utility/TypeExtensions.cs:153]: No, it was a nitpick to make the order of parameters the same as specified in the protobuf.

**Review by Francisco-Gamino — APPROVED** [review]
LGTM

**Comment by AnatoliB** [comment]
> Adding the `$RetryContext` variable will required the templates to be updated. Could you please open an issue at https://github.com/Azure/azure-functions-templates/issues for this? Thanks!

This is supposed to be an optional parameter though. I'm not sure we should be updating templates for this, as I don't expect the vast majority of users to rely on `$RetryContext`, so perhaps we should avoid adding unnecessary noise. How is this handled for Node.js functions?

**Comment by michaelpeng36** [comment]
For Node.js functions, it does not appear that updating templates was necessary.

**Pull request merged**